htlcswitch/switch.go

```diff
@@ -2215,50 +2496,31 @@ func (s *Switch) removeLink(chanID lnwire.ChannelID) ChannelLink {
 	return link
 }
 
-// UpdateShortChanID updates the short chan ID for an existing channel. This is
-// required in the case of a re-org and re-confirmation or a channel, or in the
-// case that a link was added to the switch before its short chan ID was known.
+// UpdateShortChanID locates the link with the passed-in chanID and updates the
+// underlying channel state. This is only used in zero-conf channels to allow
+// the OtherShortChannelID to be updated.
 func (s *Switch) UpdateShortChanID(chanID lnwire.ChannelID) error {
 	s.indexMtx.Lock()
 	defer s.indexMtx.Unlock()
 
-	// Locate the target link in the pending link index. If no such link
-	// exists, then we will ignore the request.
-	link, ok := s.pendingLinkIndex[chanID]
+	// Locate the target link in the link index. If no such link exists,
+	// then we will ignore the request.
+	link, ok := s.linkIndex[chanID]
 	if !ok {
 		return fmt.Errorf("link %v not found", chanID)
 	}
 
-	oldShortChanID := link.ShortChanID()
-
-	// Try to update the link's short channel ID, returning early if this
-	// update failed.
-	shortChanID, err := link.UpdateShortChanID()
+	// Try to update the link's underlying channel state, returning early
+	// if this update failed.
+	_, err := link.UpdateShortChanID()
 	if err != nil {
 		return err
 	}
 
-	// Reject any blank short channel ids.
-	if shortChanID == hop.Source {
-		return fmt.Errorf("refusing trivial short_chan_id for chan_id=%v"+
-			"live link", chanID)
-	}
-
-	log.Infof("Updated short_chan_id for ChannelLink(%v): old=%v, new=%v",
-		chanID, oldShortChanID, shortChanID)
-
-	// Since the link was in the pending state before, we will remove it
-	// from the pending link index and add it to the live link index so that
-	// it can be available in forwarding.
-	delete(s.pendingLinkIndex, chanID)
-	s.addLiveLink(link)
-
-	// Finally, alert the mail orchestrator to the change of short channel
-	// ID, and deliver any unclaimed packets to the link.
-	mailbox := s.mailOrchestrator.GetOrCreateMailBox(chanID, shortChanID)
```

Comment:
Same Q here re code removal.

Reply:
This was being called, but had no effect as the `chanID` wouldn't be in the `linkIndex`. So just changed it to update the indices, and refresh the link's channel state